stracer/src/main/scala/com/evolutiongaming/random/ThreadLocalRandom.scala

```diff
@@ -1,12 +1,24 @@
 package com.evolutiongaming.random
 
+import cats.FlatMap
 import cats.effect.{Clock, Sync}
 import cats.syntax.all._
 import com.evolutiongaming.catshelper.{ThreadLocalOf, ThreadLocalRef}
 
 object ThreadLocalRandom {
 
-  def of[F[_]: Sync: Clock: ThreadLocalOf]: F[Random[F]] = for {
+  def of[F[_]: FlatMap: Clock: ThreadLocalOf]: F[Random[F]] = {
+    val random = Random.State.fromClock[F]()
+    for {
+      random <- ThreadLocalOf[F].apply(random)
+    } yield {
+      apply(random)
+    }
+
+  }
+
+  def sync[F[_]: Sync: Clock: ThreadLocalOf]: F[Random[F]] =
```

Comment:
please name it `of + 1` = `of1` - basically incrementing version of API, this is better than making up random names :)